docs/connector-development/connector-builder-ui/tutorial.md

```diff
@@ -0,0 +1,202 @@
+
+## Setting up and testing a stream
+
+Now that you configured how to talk the API, let's set up the stream of records that will be sent to a destination later on. To do so, click the button with the plus icon next to the "Streams" section in the side menu and fill out the form:
+* Set the name to "Rates"
+* Set the "URL path" to `/exchangerates_data/latest`
+* Submit
+* Set the "Record selector" to `rates`
+
+Now the basic stream is configured and can be tested. To send test requests, suppliy testing values by clicking the "Testing values" button in top right and pasting your API key in the input.
+
+This form corresponds to what a user of this connector will need to provide when setting up a connection later on. The testing values are not saved along with the connector project.
+
+Now, click the "Test" button to trigger a test read to simulate what will happen during a sync. After a little while, you should see a single record that looks like this:
+```
+{
+  "base": "EUR",
+  "date": "2023-04-13",
+  "rates": {
+    "AED": 4.053261,
+    "AFN": 95.237669,
+    "ALL": 112.964844,
+    "AMD": 432.048005,
+    // ...
+  }
+}
+```
+
+In a real sync, this record will be passed on to a destination like a warehouse.
+
+The request/response tabs are helpful during development to see which requests and responses your connector will send and receive from the API.
+
+## Declaring the record schema
+
+Each stream of a connector needs to declare how emitted records will look like (which properties do they have, what data types will be used, ...). During a sync, this information will be passed on to the destination to configure it correctly - for example a SQL database destination can use it to properly set up the destination table, assigning the right type to each column. This process is called "normalization"
+
+By default, the stream schema is set to a simple object with unspecified properties. However, the connector builder can infer the schema based on the test read you just issued. To use the infered schema, switch to the "Detected schema" tab and click the "Import schema" button.
+
+The warning icon disappears indicating that the declared schema of your stream matches the test data.
+
+## Making the stream configurable
+
+The exchange rate API supports configuring a different base currency via request parameter - let's make this part of the user inputs that can be controlled by the user of the connector when configuring a source, similar to the API key.
+
+To do so, follow these steps:
+* Scroll down to the "Request parameters" section and add a new request parameter
+* Set the key to `base`
+* For the value, click the user icon in the input and select "New user input"
+* Set the name to "Base"
+
+Now your connector has a second configuration input. To test it, click the "Testing values" button again, set the "Base" to `USD`. Then, click the "Test" button again to issue a new test read.
+
+The record should update to use USD as the base currency:
```

Comment:
This also resulted in some diffs in my schema, such as:
<img width="463" alt="Screen Shot 2023-04-14 at 5 11 26 PM" src="https://user-images.githubusercontent.com/22731524/232172933-073dbcb2-5e2b-4553-88d2-298b1fb38b60.png">

This could be a good opportunity to explain that in this case, we want to keep the old schema since we want it to be wider, so the warning of the schemas being different should be ignored. This does also add a bit more complexity to the tutorial though so I'm also fine with leaving it out - no strong feelings.